kytos/core/interface.py

```diff
@@ -173,19 +181,303 @@ def status_reason(self):
+
+        Return False in case the tags were not able to be removed.
+        """
+        with self._tag_lock:
+            result = self.remove_tags(tags, tag_type)
+            return result
+
+    @staticmethod
+    def find_index_add(
+        available_tags: list[list[int]],
+        tag_range: list[int]
+    ) -> Optional[int]:
+        """Find the index of tags in available_tags to be added"""
+        index = bisect.bisect_left(available_tags, tag_range)
+        previous, after = False, False
+        if index - 1 > -1:
+            if tag_range[0] > available_tags[index-1][1]:
+                previous = True
+        # Out of range
+        else:
+            previous = True
+
+        if index < len(available_tags):
+            if tag_range[1] < available_tags[index][0]:
+                after = True
+        # Out of range
+        else:
+            after = True
+
+        if previous and after:
+            return index
+        return None
+
+    def add_tags(self, tags: list[int], tag_type: int) -> bool:
+        """Add tags, return True if they were added.
+        Returns False nothing nothing was added, True otherwise
+        Ensuring that ranges are not unnecessarily divided
+        available_tag e.g [[7, 10], [20, 30], [78, 92], [100, 109], [189, 200]]
+        tags examples are in each if statement.
+        """
+        available = self.available_tags[tag_type]
+        index = self.find_index_add(available, tags)
+        if index is None:
+            return False
+        if index == 0:
+            # [1, 6]
+            if tags[1] == available[index][0] - 1:
+                available[index][0] = tags[0]
+            # [1, 2]
+            else:
+                available.insert(0, tags)
+        elif index == len(available):
+            # [201, 300]
+            if available[index-1][1] + 1 == tags[0]:
+                available[index-1][1] = tags[1]
+            # [250, 300]
+            else:
+                available.append(tags)
+        else:
+            # [11, 19]
+            if (available[index-1][1] + 1 == tags[0] and
+                    available[index][0] - 1 == tags[1]):
+                available[index-1: index+1] = [
+                    [available[index-1][0], available[index][1]]
+                ]
+            # [11, 15]
+            elif available[index-1][1] + 1 == tags[0]:
+                available[index-1][1] = tags[1]
+            # [15, 19]
+            elif available[index][0] - 1 == tags[1]:
+                available[index][0] = tags[0]
+            # [15, 15]
+            else:
+                available.insert(index, tags)
+        return True
+
+    def make_tags_available(
+        self,
+        tags: list[int],
+        tag_type: int = 1
+    ) -> bool:
+        """Add a specific tag in available_tags."""
+        with self._tag_lock:
+            return self.add_tags(tags, tag_type)
+
+    @staticmethod
+    def tags_are_available(
```

Comment:
This staticmethod doesn't seem to be used anywhere, looks like it got replaced if you can double check it.

Reply:
Deleted